### DISCUSSION TITLE
Inside a Formula, how to properly refer to "/usr/local/opt/llvm@11/bin/clang"?

### DISCUSSION THREAD

**Discussion by gwerbin** [Homebrew/discussions#2218] [Writing Formulae/Casks] [answered]
I am working on packaging [C2FFI](https://github.com/rpav/c2ffi) and have successfully built it: <https://github.com/rpav/c2ffi/issues/100>.

However, I am not sure how to specify the path `/usr/local/opt/llvm@11` in the Formula. I can of course hard-code it, but that does not seems like a solution, because it makes this formula incompatible with Linux and other nonstandard installations.

This is what I have so far:

```ruby
class C2ffi < Formula
  ...

  def install
    system(
      {LDFLAGS => "-L/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib"},
      "cmake", "-S", ".", "-B", "build",
      *std_cmake_args,
      "-DBUILD_CONFIG=Release",
      "-DCMAKE_PREFIX_PATH=/usr/local/opt/llvm@11",
      -"DCMAKE_C_COMPILER=/usr/local/opt/llvm@11/bin/clang",
      "-DCMAKE_CXX_COMPILER=/usr/local/opt/llvm@11/bin/clang++",
    )
    system "cmake", "--build", "build"
    system "cmake", "--install", "build"
  end

  ...
end
```

The `/usr/local/opt/llvm@11` part is what I want to avoid hard-coding.

**Comment by carlocab** [comment, marked as answer]
Try
```ruby
Formula["llvm@11"].opt_bin/"clang"
```
You shouldn't need to set `CMAKE_PREFIX_PATH`, as `brew` sets this for you during the build based on the formula's dependencies.
